### PR TITLE
add more es6

### PR DESCRIPTION
**Pull request by denar90**
### Proposed changes
 - add more consistency of es6 instead of es5.

### CODE REVIEW

File: src/utils/deprecate.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ const deprecate = function(message, test) {
 
 deprecate._console = typeof console !== 'undefined' ? console : {};
 deprecate._warn = function() {
-  const warn = deprecate._console.warn || deprecate._console.log || function() {};
+  const warn = deprecate._console.warn || deprecate._console.log || Function.prototype;
```
Comment (JSteunou) [inline src/utils/deprecate.js]: Is `Function.prototype` es6 specific? Cause es6 new "methods" wont be translated by babel.

Comment (paulfalgout) [inline src/utils/deprecate.js]: actually in v3 since we don't support really old underscore this could change to `_.noop`

Comment (denar90) [inline src/utils/deprecate.js]: @JSteunou no) I just like `Function.prototype` more then `function() {}`. I can return how it was.

Comment (paulfalgout) [inline src/utils/deprecate.js]: Like this deprecate was pulled over from `backbone-metal` and it originally used `_.noop` until it was tested against older underscore:  https://github.com/marionettejs/backbone-metal/commit/d16fe23cb2dfe0bd9308a5e2c440c2247a9e7788#diff-2b7c460ee7b16a2db70ff8cc661ee9cdL482

File: src/collection-view.js [hunk 2]
```diff
@@ -339,7 +339,7 @@ const CollectionView = Backbone.View.extend({
 
   _sortModelsBy(models, comparator) {
     if (typeof comparator === 'string') {
-      return _.sortBy(models, function(model) {
+      return _.sortBy(models, (model) => {
```
Comment (yethee) [inline src/collection-view.js]: Here, you can use short form of arrow function:

```js
return _.sortBy(models, model => model.get(comparator));
```

**Comment by blikblum** [comment]
As i already pointed in other PR, ES6 is a superset of ES5 meaning that the old syntax is perfect valid ES6 and did not became deprecated or obsolete.

Personally i would do such changes only if there's a increase in code clarity.

IMO changing from `function(model) {` to `(model) =>` does not add in this matter. 

Anyway, unlike the PR i replied, this does not the change / increase the compiled code, so in the end is a matter of taste.

File: src/collection-view.js (hunk 2 above)
Comment (denar90) [inline src/collection-view.js]: @yethee of course, but i tried to write in one style like it was done before in other places
e.g.
```js
models = _.filter(models, (model, index) => {
        return this._shouldAddChild(model, index);
      });
```

**Comment by denar90** [comment]
It's only about consistent code.

**Comment by scott-w** [comment]
I'm in favour of this so I'll add my 👍

**Pull request merged**